lib/dl_api_lib/dl_api_lib/pivot/native/sorter.py

```diff
@@ -94,7 +107,62 @@ def _single_axis_sort(self, axis: SortAxis, directions: Sequence[OrderDirection]
         self._sort_key_list(key_list=key_list, directions=directions)
 
     def _sort_by_measure(self, axis: SortAxis, sorting_piid: int, settings: PivotMeasureSortingSettings) -> None:
-        pass  # TODO: Not implemented yet. Feeling bored? Go right ahead and contribute!
+        sorting_dim_values: Optional[tuple[DataCellVector, ...]] = None
+        other_axis = {SortAxis.columns: SortAxis.rows, SortAxis.rows: SortAxis.columns}[axis]
+
+        for idx, header in enumerate(self._pivot_dframe.iter_axis_headers(axis)):
+            if header.compare_sorting_settings(settings):
+                if sorting_dim_values is None:
+                    sorting_dim_values = header.values
+                    header.info.sorting_direction = settings.direction
+                else:  # should never actually occur, as header_values + role_spec uniquely identify sorting_idx
+                    raise exc.PivotSortingRowOrColumnIsAmbiguous()
+        if sorting_dim_values is None:
+            raise exc.PivotSortingRowOrColumnNotFound()
+
+        # First order the measure values in the selected row/column
+        # (together with their original position indices)
+        indexed_values = [
+            (old_pos_idx, value)
+            for old_pos_idx, value in enumerate(
+                self._native_dframe.iter_values_along_axis(
+                    axis=other_axis,
+                    dim_values=sorting_dim_values,
+                )
+            )
+        ]
+
+        def normalize_measure_value(value: Optional[DataCellVector]) -> Any:
+            """Convert key to a value that can be sorted natively"""
+            normalizer = self._measure_sort_strategy.get_normalizer(
+                pivot_item_id=sorting_piid, direction=settings.direction
+            )
+            return normalizer.normalize_vector_value(value)
+
+        indexed_values.sort(key=lambda pair: normalize_measure_value(pair[1]))  # pair = (old_pos_idx, value)
```

Comment:
you can initialize `normalizer` first, and then sorting key will be `lambda pair: normalizer.normalize_vector_value(pair[1])`

Reply:
and also, you can apply normalizer when you create `indexed_values`, this way if you reorder pair you won't even need a sorting key